### PR TITLE
Add weekly scheduled summary example

### PR DESCRIPTION
**Pull request by tonyhb**
This PR introduces an example scheduled function which sends a weekly
summary email to all users.

### CODE REVIEW

File: examples/weekly-summary-email/README.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+- [Usage](#usage)
+- [Code](#code)
+
+
+## Why
+
+This solves writing scheduled functions without crons or queues in seconds with the following benefits:
+
+1. Automatic retries for the entire scheduled function
+2. Splitting each user's email into its own function, for retries, local timezone sending, and logging
```
Comment (jpwilliams) [inline examples/weekly-summary-email/README.md:33]: A lovely good practice to highlight! 😍

File: examples/weekly-summary-email/README.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+## Usage
+
+<!-- A quick view of how to get started with the template. -->
+<!-- The CLI can guide them -->
+Use this quickstart with a single CLI command to get started! The CLI will then guide you through running, testing, and deploying to [Inngest Cloud](https//inngest.com/sign-up?ref=github-example).
+
+Via the CLI:
+
+```sh
+inngest-cli init --template github.com/inngest/inngest#examples/weekly-summary-email
```
Comment (jpwilliams) [inline examples/weekly-summary-email/README.md:45]:
```suggestion
inngest init --template github.com/inngest/inngest#examples/weekly-summary-email
```

File: examples/weekly-summary-email/steps/run/.env [hunk 3]
```diff
@@ -0,0 +1 @@
+SENDGRID_API_KEY=""
```
Comment (jpwilliams) [inline examples/weekly-summary-email/steps/run/.env:1]: This doesn't work for me here - the `.env` files only seem to be consumed if they're placed in the function root (`examples/weekly-summary-email`) as opposed to the step root.

Not sure if this is a bug or just not something that we thought of the first time around?

It'd be great to have it look at both locations and merge the env so we get:

- "Global" function `.env`
- Step-specific `.env` (which overwrites any function env vars with the same names)

To fix this now we can just move it up to the function root.

File: examples/weekly-summary-email/steps/run/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,15 @@
+FROM golang:1.18.2-alpine AS build
+WORKDIR /app
+COPY vendor vendor
+COPY . .
+ARG TARGETARCH
+ARG TARGETOS
+RUN --mount=type=cache,target=/root/.cache/go-build \
+    GOFLAGS=-mod=vendor GOOS=$TARGETOS GOARCH=$TARGETARCH go build -o /go/bin/main cmd/main.go
```
Comment (jpwilliams) [inline examples/weekly-summary-email/steps/run/Dockerfile:8]: Does this need to also include `mock.go`?

```suggestion
RUN --mount=type=cache,target=/root/.cache/go-build \
    GOFLAGS=-mod=vendor GOOS=$TARGETOS GOARCH=$TARGETARCH go build -o /go/bin/main cmd/main.go cmd/mock.go
```

For my benefit, is it best practice to list the exact files needed here or should we use `cmd/*.go`?

File: examples/weekly-summary-email/steps/run/cmd/main.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+		}
+
+		// Send a "summary/user.fetched" event to Inngest.  We can
+		// then create a new function that is triggered by this event
+		// to send the email vs doing it here.
+		//
+		// This gives logs per-user, allows retries per-user, and allows
+		// us to handle SendAt natively via waits within our function:
+		// https://www.inngest.com/docs/functions/step-functions#after-configuration
+		inngestgo.NewClient(os.Getenv("INNGEST_EVENT_KEY")).Send(ctx, inngestgo.Event{
```
Comment (jpwilliams) [inline examples/weekly-summary-email/steps/run/cmd/main.go:75]: Is this a secret we need to be passing in to the function? Should it be in `.env`?

File: examples/weekly-summary-email/inngest.json [hunk 6]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "Weekly summary email",
+  "id": "enough-boar-589f6a",
+  "triggers": [
```
Comment (jpwilliams) [inline examples/weekly-summary-email/inngest.json:4]:
```suggestion
{
  "name": "Weekly summary email",
  "description": "...",
  "tags": ["golang"],
  "id": "enough-boar-589f6a",
  "triggers": [
```

This needs `description` and `tags` fields adding for the site.

We could add a Dangerfile to prompt for this in the future.

File: examples/weekly-summary-email/steps/run/Dockerfile (hunk 4 above)
Comment (tonyhb) [reply examples/weekly-summary-email/steps/run/Dockerfile:8]: aha this should definitely be cmd/*.go, thanks!

**Review by jpwilliams — APPROVED** [review]

**Pull request merged**